### PR TITLE
GUI

### PR DESCRIPTION
**Pull request by Vshows**
ui

### CODE REVIEW

**Comment by Vshows** [comment]
gui

File: gui/mainwindow.cpp [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+    headimg->setPixmap(fitpixmap_userIcon);
+    layout->addWidget(headimg);
+    return layout;
+}
+
+
+void MainWindow::write_filter(){
+    ofstream ofs;
+	//!!!!文件写入位置，自定义
+    string filename = "/home/Vshows/test" ;
```
Comment (Tomahawkd) [inline gui/mainwindow.cpp]: what happens if file does not exist?

File: gui/mainwindow.cpp [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+    string ip_ori_text = ip_ori.toStdString();
+    QString ip_des = ip2->text();
+    string ip_des_text = ip_des.toStdString();
+    if(ip_ori_text.empty() || ip_des_text.empty()){
+        QMessageBox::information(this,"Failed","You type the wrong format!");
+    }
+    else{
+        string ip_text = ip_ori_text + "-"+ip_des_text+"\n";
+        ofs.open(filename,ostream::app);
+        ofs<<ip_text;
```
Comment (Tomahawkd) [inline gui/mainwindow.cpp]: what about deleted rules?

File: gui/mainwindow.cpp [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+    string filename = "/home/Vshows/test" ;
+    QString ip_ori = ip1->text();
+    string ip_ori_text = ip_ori.toStdString();
+    QString ip_des = ip2->text();
+    string ip_des_text = ip_des.toStdString();
+    if(ip_ori_text.empty() || ip_des_text.empty()){
+        QMessageBox::information(this,"Failed","You type the wrong format!");
+    }
+    else{
+        string ip_text = ip_ori_text + "-"+ip_des_text+"\n";
```
Comment (Tomahawkd) [inline gui/mainwindow.cpp]: format should be: address/mask:port(optional, default:all)

File: gui/mainwindow.cpp [hunk 4]
```diff
@@ -0,0 +1,289 @@
...
+    QLabel *headimg = new QLabel;
+    headimg->setScaledContents(true);
+    headimg->setFixedSize(160,160);
+    headimg->setPixmap(fitpixmap_userIcon);
+    layout->addWidget(headimg);
+    return layout;
+}
+
+
+void MainWindow::write_filter(){
```
Comment (Tomahawkd) [inline gui/mainwindow.cpp]: read rules also needed

**Review by Tomahawkd — CHANGES_REQUESTED** [review]
potential bugs and features required

File: gui/mainwindow.cpp (hunk 1 above)
Comment (Tomahawkd) [inline gui/mainwindow.cpp]: rule address: /var/netfilter/rules

**Review by Tomahawkd — CHANGES_REQUESTED** [review]
add requirement

**Comment by Vshows** [comment]
UI2.0

**Comment by Tomahawkd** [comment]
Now the gui is complete, waiting for file operation completion to merge

**Pull request closed without merging**